### PR TITLE
test_models: check consitency between panda and openpilot for velocity

### PR DESCRIPTION
**Pull request by jnewb1**
good check to have for angle based cars that require it as part of the safety (wanted to check for the new subarus)

### CODE REVIEW

File: selfdrive/car/tests/test_models.py [hunk 1]
```diff
@@ -345,6 +346,11 @@ def test_panda_safety_carstate(self):
       checks['brakePressed'] += brake_pressed != self.safety.get_brake_pressed_prev()
       checks['regenBraking'] += CS.regenBraking != self.safety.get_regen_braking_prev()
 
+      # Verify that panda has the correct velocity for cars that use it (angle based cars)
+      if self.CP.steerControlType in [car.CarParams.SteerControlType.angle]:
+        panda_velocity = (self.safety.get_vehicle_speed_min() + self.safety.get_vehicle_speed_max()) / 2 / VEHICLE_SPEED_FACTOR
+        checks['vEgo'] += abs(CS.vEgo - panda_velocity) > 3
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/test_models.py:352]: should also check the latest matches exactly

Comment (jnewb1) [reply selfdrive/car/tests/test_models.py:352]: any point in checking average? I'll just check last

**Comment by jnewb1** [comment]
@fredyshox want to add velocity to the body's panda safety?

**Comment by sshane** [comment]
Running on the big seg list I see at least one failure:

FAILED test_models.py::TestCarModel_2_NISSAN_LEAF_2018::test_panda_safety_carstate@CarTestRoute(route='7605de164f235364|2023-07-10--16-02-26', car_model='NISSAN LEAF 2018', segment=2) - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'vEgo': 158}

File: selfdrive/car/tests/test_models.py [hunk 2]
```diff
@@ -345,6 +346,11 @@ def test_panda_safety_carstate(self):
       checks['brakePressed'] += brake_pressed != self.safety.get_brake_pressed_prev()
       checks['regenBraking'] += CS.regenBraking != self.safety.get_regen_braking_prev()
 
+      # Verify that panda has the correct velocity for cars that use it (angle based cars)
+      if self.CP.steerControlType in [car.CarParams.SteerControlType.angle] and not self.CP.notCar:
```
Comment (sshane) [inline selfdrive/car/tests/test_models.py:350]: we've been using tuples for checks like this recently

**Comment by jnewb1** [comment]
> Running on the big seg list I see a few failures, @jnewb1 can you check it out?:
> 
> ```python
> FAILED test_models.py::TestCarModel_561_NISSAN_LEAF_2018::test_panda_safety_carstate@CarTestRoute(route='969dd380d52937c6|2023-07-18--21-02-34', car_model='NISSAN LEAF 2018', segment=1) - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'vEgo': 4}
> FAILED test_models.py::TestCarModel_1166_NISSAN_ROGUE_2019::test_panda_safety_carstate@CarTestRoute(route='b7287535e2a23992|2023-07-20--20-17-53', car_model='NISSAN ROGUE 2019', segment=1) - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'vEgo': 4}
> FAILED test_models.py::TestCarModel_1189_NISSAN_ALTIMA_2020::test_panda_safety_carstate@CarTestRoute(route='92511c3b877c7215|2023-06-18--14-47-27', car_model='NISSAN ALTIMA 2020', segment=26) - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'vEgo': 5987}
> FAILED test_models.py::TestCarModel_1379_NISSAN_LEAF_2018::test_panda_safety_carstate@CarTestRoute(route='ba9225dd390fe75e|2023-05-27--14-19-20', car_model='NISSAN LEAF 2018', segment=12) - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'vEgo': 50}
> FAILED test_models.py::TestCarModel_1448_NISSAN_LEAF_2018::test_panda_safety_carstate@CarTestRoute(route='7605de164f235364|2023-07-10--16-02-26', car_model='NISSAN LEAF 2018', segment=2) - AssertionError: 1 is not false : panda safety doesn't agree with openpilot: {'vEgo': 158}
> ```

nissan panda safety uses rear only, whereas OP uses front and rear

**Pull request merged**